### DISCUSSION TITLE
How can I create entities more than 1048575?

### DISCUSSION THREAD

**Discussion by sawino** [skypjack/entt#1087] [Q&A] [answered]
I'm trying to create 100,000,000 entities but the creation failed when calls registry.create() for more than 1,048,575 times.
`
    entt::registry r;

    // 100,000,000 entities
    for (size_t i = 0ul; i < 100000000ul; i++) {
        // failed when i is greater than 1048575
        auto entity = r.create();
    }
`

The assertion failure happened at entt::basic_storage::entity_at(). 

`
   auto entity_at(const std::size_t pos) const noexcept {
        // exception happens here
        ENTT_ASSERT(pos < underlying_type::traits_type::to_entity(null), "Invalid element");
        return underlying_type::traits_type::combine(static_cast<typename underlying_type::traits_type::entity_type>(pos), {});
    }

`
I tried to debug and read the documents, but I cannot find a solution. How can I achieve my goal? Do I miss any configurations?
Any comments will be appreciated.

**Comment by Daedie-git** [comment]
By default, entt represents entities using 32 bit. 20 bit for the id (max value: 1048575) and 12 bit for the version/generation. This can be customized to suit your needs.

**Reply by sawino** [reply]
Thanks for your message which helped me to understand the magic number, and explained how does it work.

**Comment by skypjack** [comment, marked as answer]
https://github.com/skypjack/entt/wiki/Frequently-Asked-Questions#custom-entity-identifiers-yay-or-nay

**Reply by sawino** [reply]
Thanks for the reply. I have resolved my issue by changing the ENTT_ID_TYPE from uint32_t to uint64_t in the entt/config/config.h.

Code

`
\#ifndef ENTT_ID_TYPE
\#    include \<cstdint\>
\#    define ENTT_ID_TYPE std::uint64_t
\#endif

`
